### PR TITLE
Add GitHub workflow to build docs & make them viewable in CircleCI

### PR DESCRIPTION
**Pull request by romanlutz**
WIP, no need to review yet, but I need to run the new GitHub Action/CircleCI workflow.

Signed-off-by: Roman Lutz <rolutz@microsoft.com>

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# This CircleCI workflow builds our documentation into HTML pages.
+# The generated pages are viewable as artifacts in CircleCI.
+# CircleCI can be run on PRs from forks & allows viewing stored artifacts
+# without download.
```
Comment (riedgar-ms) [inline .circleci/config.yml:4]: Do you want to call out the PR-from-fork bit is not possible from ADO?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      - run:
+          name: 'Install required packages'
+          command: sudo pip install -r requirements.txt
+      
+      - run:
+          name: 'Build'
+          command: python -m sphinx -b html -j auto docs docs/_build/html
+
+      - store_artifacts:
+          path: docs/_build/html
```
Comment (riedgar-ms) [inline .circleci/config.yml:31]: So are these visible now? If so, what's the URL?

Comment (romanlutz) [reply .circleci/config.yml:31]: https://11-133444044-gh.circle-artifacts.com/0/docs/_build/html/index.html

I'd like it to be a single click from the PR view, so there's probably some extra change coming once I figure out how to add that link directly into the PR the way scikit-learn does. Right now it's a handful of clicks starting with navigating to CircleCI (see "build-webpage" in the checks and then search for the artifacts, find the one with index.html and click on it).

The link at the top should make it clear what this is for, though, I hope :-)

File: .circleci/config.yml (hunk 1 above)
Comment (romanlutz) [reply .circleci/config.yml]: done

**Comment by romanlutz** [comment]
> 
> 
> A comment update request, and.... are these visible at some URL?

Yes! See the response on your other comment :-)

**Comment by Superleah06** [comment]
> WIP, no need to review yet, but I need to run the new GitHub Action/CircleCI workflow.
> 
> Signed-off-by: Roman Lutz [rolutz@microsoft.com](mailto:rolutz@microsoft.com)

[
`
](url)`

**Pull request merged**